app/src/main/java/com/gh4a/adapter/timeline/ReviewViewHolder.java

```diff
@@ -0,0 +1,298 @@
+    }
+
+    @Override
+    public void bind(TimelineItem.TimelineReview item) {
+        Review review = item.review;
+        mShowDetailsButton.setTag(review);
+
+        AvatarHandler.assignAvatar(mAvatarView, review.getUser());
+        mAvatarContainer.setTag(review.getUser());
+
+        formatTitle(review);
+
+        boolean hasBody = !TextUtils.isEmpty(review.getBody());
+        if (hasBody) {
+            mImageGetter.bind(mBodyView, review.getBodyHtml(), review.getId());
+            mBodyView.setVisibility(View.VISIBLE);
+        } else {
+            mBodyView.setVisibility(View.GONE);
+        }
+
+        if (mCallback.canQuote()) {
+            mBodyView.setCustomSelectionActionModeCallback(
+                    new UiUtils.QuoteActionModeCallback(mBodyView) {
+                        @Override
+                        public void onTextQuoted(CharSequence text) {
+                            mCallback.quoteText(text);
+                        }
+                    });
+        } else {
+            mBodyView.setCustomSelectionActionModeCallback(null);
+        }
+
+        boolean hasDiffs = !item.getDiffHunks().isEmpty();
+        if (mDisplayReviewDetails && hasDiffs) {
+            ArrayList<TimelineItem.Diff> diffHunks = new ArrayList<>(item.getDiffHunks());
+            LayoutInflater inflater = LayoutInflater.from(mContext);
+            Map<String, FileDetails> files = new HashMap<>();
+
+            int viewIndex = 0;
+            for (TimelineItem.Diff diffHunk : diffHunks) {
+                CommitComment commitComment = diffHunk.getInitialComment();
+                String filename = commitComment.getPath();
+                int commentCount = diffHunk.comments.size();
+                boolean isOutdated = commitComment.getPosition() == -1;
+
+                if (files.containsKey(filename)) {
+                    FileDetails details = files.get(filename);
+                    details.isOutdated = details.isOutdated && isOutdated;
+                    details.count += commentCount;
+                    continue;
+                }
+
+                View row = mDetailsContainer.getChildAt(viewIndex);
+                if (row == null) {
+                    row = inflater.inflate(R.layout.row_timeline_review_file_details,
+                            mDetailsContainer, false);
+                    mDetailsContainer.addView(row);
+                    row.setOnClickListener(this);
+                }
+                row.setTag(review);
+                row.setTag(R.id.review_comment_id, commitComment.getId());
+
+                files.put(filename, new FileDetails(row, isOutdated, commentCount));
+
+                viewIndex += 1;
+            }
+
+            for (Map.Entry<String, FileDetails> detailsEntry : files.entrySet()) {
+                FileDetails fileDetails = detailsEntry.getValue();
+                TextView tvFile = (TextView) fileDetails.row.findViewById(R.id.tv_file);
```

Comment:
There would be no real benefit to that as these FileDetails are created inside of bind method anyway. But perhaps we could store them somewhere and in bind method update their data or create new only when necessary.